src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -345,6 +345,7 @@ private string GetDefaultAuthorityInstance()
 
             if (!string.IsNullOrWhiteSpace(Config.Instance))
             {
+                Config.Instance = Config.Instance.TrimEnd(' ', '/');
```

Comment:
>Config.Instance = Config.Instance.TrimEnd(' ', '/'); [](start = 16, length = 52)

any reason we are not asking the developer to specify this correctly? exception?

Reply:
Our code later on assumes that this has no / for joining, and when we parse we split on "/".  So this is for us to not screw up.  It's legitimate syntax for the developer to supply the URI as https://login.microsoftonline.com/ or https://login.microsoftoneline.com (no trailing slash).  So this code just makes sure we do the right thing later on when combining these URI paths and parsing them.  

For example, Path.Combine actually solves this for file system paths and does the right thing around making sure there's only one / or \ delimiter.  But new Uri() does NOT do this and so we can end up with https://login.microsoftonline.com//tenantid and then when we parse out the tenant id (by doing a split on '/') we get the empty string for tenant. 

So, bottom line, I don't think we should throw an exception here, we're just keeping values tidy for later processing internally.

Reply:
ok. thanks for explaining. 

---
In reply to: [295450204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1224#discussion_r295450204) [](ancestors = 295450204)